katas/content/solving_graph_coloring/read_coloring/Verification.qs

```diff
@@ -0,0 +1,52 @@
+namespace Kata.Verification {
+    open Microsoft.Quantum.Diagnostics;
+    open Microsoft.Quantum.Arrays;
+    open Microsoft.Quantum.Convert;
+    open Microsoft.Quantum.Katas;
+
+    @EntryPoint()
+    operation CheckSolution() : Bool {
+        for nBits in 1..3 {
+            for V in 1..3 {
+                use qs = Qubit[V * nBits];
+                for state in 0..(1 <<< (V * nBits)) - 1 {
+                    // Prepare the register in the input state
+                    let binaryState = IntAsBoolArray(state, V * nBits);
+                    ApplyPauliFromBitString(PauliX, true, binaryState, qs);
```

Comment:
Would a `within apply` construct make sense here too?

Reply:
Same as above, for very simple apply code doesn't improve the code much